### PR TITLE
admin repos view: make stats numbers clickable

### PR DESCRIPTION
**Pull request by mrnugget**
This makes the numbers on the top of the site-admin "Repositories" page clickable, if there is a corresponding filter.

This was requested by @ryphil. 

@eseliger contributed the changes to `FilteredConnection` here.

## Demo video

https://user-images.githubusercontent.com/1185253/198313569-9e7b46c5-1888-4dab-98c8-5db6d982d582.mp4

## Test plan

- Manually tested

## App preview:

- [Web](https://sg-web-mrn-clickable-legend.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.01% (+2.08 kb) | 0.02% (+2.08 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/mrn/clickable-legend/compare-report.html;tab=live_object) for a full comparison between the commits 9cef7d5 and 43c711a or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Review by umpox — APPROVED** [review]
Nice!

File: client/web/src/components/FilteredConnection/FilteredConnection.tsx [hunk 1]
```diff
@@ -430,6 +430,23 @@ export class FilteredConnection<
                     )
                 )
         )
+
+        // React to location changes.
+        this.subscriptions.add(
```
Comment (eseliger) [inline client/web/src/components/FilteredConnection/FilteredConnection.tsx:435]: I wasn't able to test it, so I hope this cannot trigger inf render loops or so!

**Review by eseliger — APPROVED** [review]

File: client/web/src/components/FilteredConnection/FilteredConnection.tsx (hunk 1 above)
Comment (mrnugget) [reply client/web/src/components/FilteredConnection/FilteredConnection.tsx:435]: Tested for that case. No, there are no infinite render loops.

**Review by varsanojidan — APPROVED** [review]
Looks great!

**Pull request merged**